src/modules/kurtosis/base.cc

```diff
@@ -0,0 +1,128 @@
+            config.debugMode,
+            getInputBuffer().shape().numberOfAspects(),
+            getInputBuffer().shape().numberOfFrequencyChannels(),
+            getInputBuffer().shape().numberOfTimeSamples(), 
+            getInputBuffer().shape().numberOfPolarizations(),
+            config.numberOfKurtosisStddev,
+            config.kurtosisChannelLength
+        )
+    );
+
+    if constexpr (!std::is_same<IT, OT>::value) {
+        BL_FATAL("This module requires the type of the input "
+                 "({}) and output ({}) to be the same.",
+                 TypeInfo<IT>::name, TypeInfo<OT>::name); 
+        BL_INFO("Contact the maintainer if this "
+                "functionality is required.");
+        BL_CHECK_THROW(Result::ERROR);
+    }
+
+    // Link output buffer or link input with output.
+    BL_CHECK_THROW(Link(output.buf, input.buf));
+   
+    // 32 comes from 8192 / 256
+    // we are deciding to do blocks of 256 for kurtosis
+    this->output.mask = ArrayTensor<Device::CUDA, U8>({config.numberOfMaskRuns, 
+            getInputBuffer().shape().numberOfAspects(),
+            getInputBuffer().shape().numberOfFrequencyChannels(),
+            getInputBuffer().shape().numberOfTimeSamples() / (4 * config.kurtosisChannelLength)
+            }, true);
+
+    this->maskCounter = 0;
+    this->maskOutFile.open(std::string(this->config.maskFilePath), std::ios::binary);
+
+    // Print configuration values.
+    BL_INFO("Type: {} -> {}", TypeInfo<IT>::name, TypeInfo<OT>::name);
+    BL_INFO("Shape: {} -> {}", getInputBuffer().shape(), 
+                              getOutputBuffer().shape());
+    BL_INFO("Config: debugMode:          {}", this->config.debugMode);
+    BL_INFO("Config: kurtosisChannelLength:  {}", this->config.kurtosisChannelLength);
+    BL_INFO("Config: numberOfKurtosisStddev:  {}", this->config.numberOfKurtosisStddev);
+    BL_INFO("Config: numberOfMaskRuns:       {}", this->config.numberOfMaskRuns);
+    BL_INFO("Config: maskFilePath:           {}", this->config.maskFilePath);
+    BL_INFO("Output Mask Shape: {}", getOutputMask().shape());
+}
+
+template<typename IT, typename OT>
+Result Kurtosis<IT, OT>::writeMaskToDisk() {
+    // mask write implementation
+
+    int masksize = config.numberOfMaskRuns * getInputBuffer().shape().numberOfAspects() * getInputBuffer().shape().numberOfFrequencyChannels() * 8;
+        
+    this->maskOutFile.write(reinterpret_cast<const char*>((this->output.mask.data())), masksize);
+
+    return Result::SUCCESS;
+}
+
+template<typename IT, typename OT>
+Kurtosis<IT, OT>::~Kurtosis() {
+    // destructor
+    if (this->maskCounter == config.numberOfMaskRuns) {
+        this->writeMaskToDisk();
+    }
+    this->maskOutFile.close();
+}
+
+template<typename IT, typename OT>
+Result Kurtosis<IT, OT>::process(const U64& currentStepCount, const Stream& stream) {
+
+    BL_CHECK(Link(output.buf, input.buf));
+
+
+    if (this->maskCounter == config.numberOfMaskRuns) {
```

Comment:
Configurable no-write scenario... maybe if the maskFilePath is "", zero length string...